src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1437,4 +1450,122 @@ by rw [←finset.mem_coe, coe_finset_walk_length_eq]
 
 end walk_counting
 
+section bridge_edges
+
+/-! ### Bridge edges -/
+
+/-- An edge of a graph is a *bridge* if, after removing it, its incident vertices
+are no longer reachable from one another. -/
+def is_bridge (G : simple_graph V) (e : sym2 V) : Prop :=
+e ∈ G.edge_set ∧
+sym2.lift ⟨λ v w, ¬ (G.delete_edges {e}).reachable v w, by simp [reachable_comm]⟩ e
+
+lemma is_bridge_iff {u v : V} :
+  G.is_bridge ⟦(u, v)⟧ ↔ G.adj u v ∧ ¬ (G.delete_edges {⟦(u, v)⟧}).reachable u v := iff.rfl
+
+lemma reachable_delete_edges_iff_exists_walk {v w : V} :
+  (G.delete_edges {⟦(v, w)⟧}).reachable v w ↔ ∃ (p : G.walk v w), ¬ ⟦(v, w)⟧ ∈ p.edges :=
+begin
+  split,
+  { rintro ⟨p⟩,
+    use p.map (hom.map_spanning_subgraphs (G.delete_edges_le _)),
+    simp_rw [walk.edges_map, list.mem_map, hom.map_spanning_subgraphs_apply, sym2.map_id', id.def],
+    rintro ⟨e, h, rfl⟩,
+    simpa using p.edges_subset_edge_set h, },
+  { rintro ⟨p, h⟩,
+    exact ⟨p.to_delete_edge _ h⟩, },
+end
+
+lemma is_bridge_iff_adj_and_forall_walk_mem_edges {v w : V} :
+  G.is_bridge ⟦(v, w)⟧ ↔ G.adj v w ∧ ∀ (p : G.walk v w), ⟦(v, w)⟧ ∈ p.edges :=
+begin
+  rw is_bridge_iff,
+  apply iff.and iff.rfl,
+  rw [reachable_delete_edges_iff_exists_walk, not_exists_not],
+end
+
+lemma reachable_delete_edges_iff_exists_cycle.aux [decidable_eq V]
+  {u v w : V}
+  (hb : ∀ (p : G.walk v w), ⟦(v, w)⟧ ∈ p.edges)
+  (c : G.walk u u)
+  (hc : c.is_trail)
+  (he : ⟦(v, w)⟧ ∈ c.edges)
+  (hw : w ∈ (c.take_until v (c.fst_mem_support_of_mem_edges he)).support) :
+  false :=
```

Comment:
The point of this argument is to create a contradiction, no single hypothesis is the one that's negated. This also breaks the proof since then I need an artificial `intro hw` at the beginning.